### PR TITLE
client/setec: export a Refresh method on the Store

### PR DESCRIPTION
**Pull request by creachadair**
(A proposal for discussion)

Allow the program to explicitly check for new secret values at a specific
moment, e.g. in response to a signal or other operator-controlled input.

### CODE REVIEW

File: client/setec/store.go [hunk 1]
```diff
@@ -241,6 +241,26 @@ func (s *Store) Close() error {
...
 }
 
+// Refresh synchronously checks for new versions of all the secrets currently
+// known by s. It blocks until the refresh is complete or until ctx ends.
+//
+// Updates are managed automatically when a Store is created and by the polling
+// mechanism, but a caller may invoke Refresh directly if it wants to check for
+// new secret values at a specific moment.
+func (s *Store) Refresh(ctx context.Context) error {
+	s.countPolls.Add(1)
```
Comment (danderson) [inline client/setec/store.go:251]: Should refreshes be x/singleflighted? With this API, a naively written client could generate a flurry of server work (and audit logs?) for something that could be coalesced gracefully.

**Review by danderson — APPROVED** [review]
This seems fine implementation-wise, but why? Did you come across a need/use for this? Not an objection, just my default setting of keeping APIs small and not having a clear idea of why I'd use this.

File: client/setec/store.go (hunk 1 above)
Comment (creachadair) [reply client/setec/store.go:251]: Not a bad idea. I figured a client could do that anyway, if they just grab the client and spam the server whenever they want a new secret value, but there wouldn't be any harm in adding a bit of extra politeness here. Will do.

**Comment by creachadair** [comment]
> This seems fine implementation-wise, but why? Did you come across a need/use for this? Not an objection, just my default setting of keeping APIs small and not having a clear idea of why I'd use this.

I don't have an immediate need, the thought came to me as I was writing up the docs, in which my interlocutor across the corpus callosum said, "ok, you said we have this polling mechanism so a server doesn't have to be restarted when a new value comes around, but what happens when I have to _choose_ when to rotate the secret? Sure, maybe I can just leave them both lying around for a while, and my once-a-day check will get it, but if I need to act quickly that means my only option is to reboot the whole service again."

So then I went looking, and noticed that the code is actually trivially and safely usable by an external caller (though, as you noted, we probably ought to single-flight it), so apart from the extra surface area, it seemed like a nice story to tell in the docs. (Although, having said that, I haven't actually put it into the docs yet because I wanted to have this conversation first 🙂)

That said: I don't think it's immediately necessary yet.

**Comment by danderson** [comment]
Yeah, seems reasonable to me. I would actually want to hook this to a /debug handler in most of our stuff tbh, alongside some misc debug info about the active store. Technically that only needs an internal handler if the client owns the debug handlers, but exposing refresh control isn't terrible, I could see someone wanting to do the thing I suppose.

**Comment by creachadair** [comment]
A /debug handler is a good idea, I hadn't even thought of that part. I was thinking more of SIGHUP handlers or maybe (eventually) a webhook from the secrets server saying "I gots stuff for you".

**Pull request merged**